source/Calamari.Shared/Deployment/SpecialVariables.cs

```diff
@@ -15,6 +18,15 @@ public static string GetLibraryScriptModuleName(string variableName)
             return variableName.Replace("Octopus.Script.Module[", "").TrimEnd(']');
         }
 
+        public static ScriptSyntax GetLibraryScriptModuleLangauge(VariableDictionary variables, string variableName)
+        {
+            var expectedName = variableName.Replace("Octopus.Script.Module[", "Octopus.Script.Module.Language[");
+            var syntaxVariable = variables.GetNames().FirstOrDefault(x => x == expectedName);
```

Comment:
You know, I'm not convinced about that as an approach. (I know that's whats being talked about).
I'm really a fan of the capabilities approach, especially when we think about things like pre-preemptively showing a warning "your target needs to have powershell 5.1 on it", before executing a script, rather than during executing.
I also think the user has the most context on what script they want to run, and what language they want to use.